ocaml/xapi/xapi_clustering.ml

```diff
@@ -164,6 +164,27 @@ let with_clustering_lock_if_cluster_exists ~__context where f =
   | _ ->
       with_clustering_lock where f
 
+let find_cluster_host_by_network ~__context ~addr =
+  match
+    Db.Cluster_host.get_all ~__context
+    |> List.filter (fun cluster_host ->
+           let pIF = Db.Cluster_host.get_PIF ~__context ~self:cluster_host in
+           let (Cluster_interface.IPv4 ip) =
+             ip_of_pif (pIF, Db.PIF.get_record ~__context ~self:pIF)
```

Comment:
The DB calls here are also done for each host in `nodel` again and again. It is worth creating a list of addresses with cluster_host refs once, and then have the find function use that.